File: src/main/java/com/sfh/pokeRogueBot/model/decisions/PossibleAttackMove.java
package com.sfh.pokeRogueBot.model.decisions;

import com.sfh.pokeRogueBot.model.enums.MoveTargetAreaType;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class PossibleAttackMove {

    private final int index;
    private final int minDamage;
    private final int maxDamage;
    private final int attackPriority;
    private final int attackerSpeed;
    private final String attackName;
    private final MoveTargetAreaType targetAreaType;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/decisions/PossibleAttackMove.kt
package com.sfh.pokeRogueBot.model.decisions

import com.sfh.pokeRogueBot.model.enums.MoveTargetAreaType
import com.sfh.pokeRogueBot.model.enums.PokeType

data class PossibleAttackMove(
    val index: Int,
    val minDamage: Int,
    val maxDamage: Int,
    val attackPriority: Int,
    val attackerSpeed: Int,
    val attackName: String,
    val targetAreaType: MoveTargetAreaType,
    val attackType: PokeType
)